lib/finch.ex

```diff
@@ -152,23 +154,31 @@ defmodule Finch do
   end
 
   def request(name, method, url, headers \\ [], body \\ nil, opts \\ []) do
-    uri = URI.parse(url)
+    {uri, shp} = parse_and_normalize_url(url)
 
     req = %{
-      scheme: normalize_scheme(uri.scheme),
+      scheme: uri.scheme,
       host: uri.host,
       port: uri.port,
       method: build_method(method),
-      path: uri.path || "/",
+      path: uri.path,
       headers: headers,
       body: body,
       query: uri.query
     }
 
-    pool = PoolManager.get_pool(name, req.scheme, req.host, req.port)
+    pool = PoolManager.get_pool(name, shp)
     Pool.request(pool, req, opts)
   end
 
+  defp parse_and_normalize_url(url) when is_binary(url) do
+    uri = URI.parse(url)
+    uri = %URI{uri | scheme: normalize_scheme(uri.scheme), path: uri.path || "/"}
```

Comment:
The type of `URI.scheme` is `nil | binary()` so setting it to an atom here breaks that contract. We should probably just build a map with the extracted info, and return that. 

I'm not really a fan of the tuple we are returning either. What do you think about returning ok or error tuples from this function? We can let the caller build a {s, h, p} tuple from the map returned on :ok, and on :error, we could let the error bubble up in `pool_options!/1`, where it will be raised and annotated with a bit more information. In `request/3,4,5` I think it would be better to return the error tuple and avoid raising. What do you think?

Reply:
That sounds good to me! Nice catch with the scheme, too :)